doc/stdenv/stdenv.chapter.md

```diff
@@ -317,6 +317,61 @@ The script will be usually run from the root of the Nixpkgs repository but you s
 
 For information about how to run the updates, execute `nix-shell maintainers/scripts/update.nix`.
 
+### Recursive attributes in `mkDerivation`
+
+If you pass a function to `mkDerivation`, it will receive as its argument the final arguments, considering use of `overrideAttrs`. For example:
+
+```nix
+mkDerivation (finalAttrs: {
+  pname = "hello";
+  withFeature = true;
+  configureFlags =
+    lib.optionals finalAttrs.withFeature ["--with-feature"];
+})
```

Comment:
> and in the future hopefully we can slap lib.makeOverridable in the fetchers so I can override their args instead of the resulting derivation

See relevant https://github.com/NixOS/nixpkgs/pull/158018

Reply:
Let's scope this out and follow it up in #158018.